Library/Homebrew/test/patch_spec.rb

```diff
@@ -0,0 +1,165 @@
+require "patch"
+
+describe Patch do
+  describe "#create" do
+    context "simple patch" do
+      subject { described_class.create(:p2, nil) }
+      it { is_expected.to be_kind_of ExternalPatch }
+      it { is_expected.to be_external }
+      its(:strip) { is_expected.to eq(:p2) }
+    end
+
+    context "string patch" do
+      subject { described_class.create(:p0, "foo") }
+      it { is_expected.to be_kind_of StringPatch }
+      its(:strip) { is_expected.to eq(:p0) }
+    end
+
+    context "string patch without strip" do
+      subject { described_class.create("foo", nil) }
+      it { is_expected.to be_kind_of StringPatch }
+      its(:strip) { is_expected.to eq(:p1) }
+    end
+
+    context "data patch" do
+      subject { described_class.create(:p0, :DATA) }
+      it { is_expected.to be_kind_of DATAPatch }
+      its(:strip) { is_expected.to eq(:p0) }
+    end
+
+    context "data patch without strip" do
+      subject { described_class.create(:DATA, nil) }
+      it { is_expected.to be_kind_of DATAPatch }
+      its(:strip) { is_expected.to eq(:p1) }
+    end
+
+    it "raises an error for unknown values" do
+      expect {
+        described_class.create(Object.new)
+      }.to raise_error(ArgumentError)
```

Comment:
I'd put `described_class.create(Object.new)` on the same line if it's < 80 chars.